indexing-service/src/main/java/org/apache/druid/indexing/overlord/HeapMemoryTaskStorage.java

```diff
@@ -183,7 +183,7 @@ public void setStatus(TaskStatus status)
     try {
       final ImmutableList.Builder<Task> listBuilder = ImmutableList.builder();
       for (final TaskStuff taskStuff : tasks.values()) {
-        if (taskStuff.getStatus().isRunnable()) {
+        if (taskStuff.getStatus().isRunnable() && taskStuff.getTask() != null) {
```

Comment:
I guess `task` is never null in HeapMemoryTaskStorage?

Reply:
Good point :+1: